pyproject.toml

```diff
@@ -3,7 +3,7 @@ requires = [
     "setuptools>=42",
     "wheel",
     "cmake",
-    "numpy",
+    "numpy<1.24",
 ]
 build-backend = "setuptools.build_meta"
```

Comment:
The latest pip versions have broken the legacy --no-build-isolation behavior that implied that the source is built under the "build" directory in the source tree. 

This broke incremental builds on `-e` installations, since the the object files in `/tmp` are deleted. One can get the old behavior back by passing `--no-use-pep517` to pip, which doesn't work because of this line. What is the reason to use this specific backend?

Reply:
Why does upgrading pip has anything to do with this restriction to use numpy<1.24? What you mention about `build` is an issue, but it seems orthogonal to this. See https://github.com/impy-project/impy/pull/90#issuecomment-1359030501

If you install numpy-1.24, the build on CI breaks, @jncots discovered this. https://github.com/impy-project/impy/pull/90#issuecomment-1359080327

Reply:
I meant the build-backend line

Reply:
I don't know how to fix this. On CI, setuptools still works. I also want incremental builds so if you have an idea how to fix this, please go ahead. It is not related to this PR, though, so I would prefer if we discuss this in an issue. I did not change the build-backend in this PR.

Reply:
@afedynitch Is this resolved now by your change of commenting out build-backend?

Reply:
Yes! :)